src/main/java/com/cleanroommc/groovyscript/compat/mods/advancedmortars/Mortar.java

```diff
@@ -69,16 +75,73 @@ public void remove(RecipeMortar recipe) {
         addBackup(recipe);
     }
 
-    public static List<EnumMortarType> getTypes(RecipeMortar value) {
-        List<EnumMortarType> list = new ArrayList<>();
-        ((RegistryRecipeMortarAccessor) MortarAPI.RECIPE_REGISTRY)
+    public static List<EnumMortarType> getTypes(RecipeMortar recipe) {
+        return ((RegistryRecipeMortarAccessor) MortarAPI.RECIPE_REGISTRY)
                 .getRecipeMap()
-                .forEach((key, value1) -> value1
-                        .stream()
-                        .filter(recipe -> recipe == value)
-                        .map(recipe -> key)
-                        .forEach(list::add));
-        return list;
+                .entrySet()
+                .stream()
+                .filter(x -> x.getValue().contains(recipe))
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toList());
+    }
+
+    public class RecipeBuilder extends AbstractRecipeBuilder<RecipeMortar> {
+
+        private final List<String> types = new ArrayList<>();
+        private int duration;
+        private ItemStack secondaryOutput;
+        private float secondaryOutputChance;
+
+        public RecipeBuilder type(String type) {
+            this.types.add(type);
+            return this;
+        }
+
+        public RecipeBuilder duration(int duration) {
+            this.duration = duration;
+            return this;
+        }
+
+        public RecipeBuilder secondaryOutput(ItemStack itemStack) {
+            this.secondaryOutput = itemStack;
+            return this;
+        }
+
+        public RecipeBuilder secondaryOutput(ItemStack itemStack, float chance) {
+            this.secondaryOutput = itemStack;
+            this.secondaryOutputChance = chance;
```

Comment:
since you can assign the output without chance being set, it seems reasonable to also let chance be set without a itemstack.